internal/generator/vector/output/loki/loki.go

```diff
@@ -206,26 +207,33 @@ func Tenant(l *logging.Loki) Element {
 }
 
 func TLSConf(o logging.OutputSpec, secret *corev1.Secret, op Options) []Element {
+	conf := []Element{}
 	if isDefaultOutput(o.Name) {
 		// Set CA from logcollector ServiceAccount for internal Loki
 		tlsConf := security.TLSConf{
 			ComponentID: strings.ToLower(vectorhelpers.Replacer.Replace(o.Name)),
 			CAFilePath:  `"/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt"`,
 		}
 		tlsConf.SetTLSProfileFromOptions(op)
-		return []Element{
-			tlsConf,
-		}
+		return append(conf, tlsConf)
 	}
+
 	if o.Secret != nil || (o.TLS != nil && o.TLS.InsecureSkipVerify) {
 
 		if tlsConf := security.GenerateTLSConf(o, secret, op, false); tlsConf != nil {
 			tlsConf.NeedsEnabled = false
-			return []Element{tlsConf}
+			conf = append(conf, tlsConf)
+		}
+	} else if secret != nil {
+		// Use secret of logcollector service account as backup
+		tlsConf := security.TLSConf{
```

Comment:
I think the logic is missing something here because it's not using any of the helper functions.  
* What if my secret defines a CA to use?  
* What if I define a secret ... but I don't get any TLS Profile options here

Following "GenerateTLSConf" leads me to think it still nees to utilize that function... and then check the CA afterwards and set it like this if not defined